src/encode.cpp

```diff
@@ -117,6 +117,63 @@ static mlir::Type getTensorElemTy(mlir::Value v) {
   return v.getType().dyn_cast<mlir::TensorType>().getElementType();
 }
 
+static pair<Tensor, Tensor> 
+getBroadcastTensor(State &st, mlir::Value arg0, mlir::Value arg1) {
+  // reference: https://numpy.org/doc/stable/user/basics.broadcasting.html
+  auto ty0 = arg0.getType().cast<mlir::RankedTensorType>();
+  auto ty1 = arg1.getType().cast<mlir::RankedTensorType>();
+  auto resRank = max(ty0.getRank(), ty1.getRank());
+  auto inVars0 = Index::boundIndexVars(resRank);
+  auto inVars1 = Index::boundIndexVars(resRank);
+
+  vector<Expr> outVars0, outVars1, resDims;
+  for(int i = 0; i < resRank; i++) {
+    auto idx0 = ty0.getRank() - 1 - i;
+    auto idx1 = ty1.getRank() - 1 - i;
+
+    if(idx0 >= 0 && idx1 >= 0) {
+      auto d1 = ty0.getDimSize(idx0);
+      auto d2 = ty1.getDimSize(idx1);
+      assert(d1 == 1 || d2 == 1 || d1 == d2);
```

Comment:
What if `ty0.getDimSize(idx0)` has an unknown size?

Reply:
hmm... I've searched the test suite but I can't find tosa.add with ? dimension. fixed to return the original tensor if there is an unknown dim.